active/0004-extensibility-support-for-deployment-stacks.md

```diff
@@ -0,0 +1,405 @@
+these scenarios. One way to mitigate the problem in the UX directly is to provide a non-error diagnostic on the extension
+line for extensions that require configuration so at least the user is aware at the time of authoring.
+
+## Alternatives
+
+### Resource deployment parameters
+
+The reason why extension configuration is moved from a template to the deployment input is mostly due to needing to be able to
+reference ARM resources in a parent deployment as the source of truth for the configuration. It's not currently possible 
+to define a deployment parameter as a reference to an ARM resource in a way that can be analyzed at template evaluation time. 
+There are draft designs for this type of feature, but it is not available at the start time of this document. Deployment
+parameters are currently limited to primitives and basic types that do not provide enough engine-level metadata to make 
+assertions about the source of a parameter value (for example, how could it be known that a secure string parameter is 
+always sourced from the AKS cluster admin credentials in the Kubernetes example?).
+
+Resource deployment parameters, if designed in a way that enforces a certain resource type, would solve this issue, and
+there wouldn't be need to move the extension configuration definition site. Having this available would also solve the
+root deployment problem because the parameter would be like any other parameter, but enforced to be a string resource ID
+of the defined resource type. This would also minimize the changes needed to clients.
+
+The reason this approach is not chosen is due to dependency on this feature development which would delay stacks extension
+support. It needs to be decided if it's worth implementing that feature first.
+
+Here is an example of this approach:
+
+```bicep
+param aksCluster Resource<'Microsoft.ContainerService/managedClusters@2024-02-01'> // = "/subscriptions/.../resourceGroups/.../Microsoft.ContainerService/managedClusters/...
+
+param namespace string = 'default'
+
+extension kubernetes with {
+  kubeConfig: aksCluster.listClusterAdminCredential().kubeconfig[0].value
+  namespace: namespace
+}
+
+// ...
+```
+
+### Use of existing resources
+
+The extension configuration could be kept as-is but restricted to existing resource reference expressions. There should
+not be a need to include this resource in the deployment GET response for stacks usage because it will be compiled to
+the new configuration format and embedded in that.
+
+Here is an example:
+```bicep
+param clusterName string
+param namespace string = 'default'
+
+// duplicates a resource definition, but leaves the extension config definition site unchanged
+resource aksCluster 'Microsoft.ContainerService/managedClusters@2024-02-01' existing = {
+  name: clusterName
+}
+
+extension kubernetes with {
+  kubeConfig: aksCluster.listClusterAdminCredential().kubeconfig[0].value
```

Comment:
Shouldn't this be blocked? Line 145 says that references to deployment parameters are prohibited in extension configurations, but the `aksCluster.listClusterAdminCredential()` will be compiled to `[(listClusterAdminCredential(resourceId('Microsoft.ContainerService/managedClusters', parameters('clusterName')))...]`

Reply:
In this case, the engine could evaluate this parameter because it's not a secure parameter and persist the value inside the `ArmApiCall` directive in the `resourceId` shown in the ARM template example. It's an alternative that would allow parameter use in this case because it only depends on the name of the resource and scope detail.

Reply:
I updated the design to include language expression parsing.